nextcord/http.py

```diff
@@ -155,25 +154,255 @@ def bucket(self) -> str:
         return f"{self.channel_id}:{self.guild_id}:{self.path}"
 
 
-class MaybeUnlock:
-    def __init__(self, lock: asyncio.Lock) -> None:
-        self.lock: asyncio.Lock = lock
-        self._unlock: bool = True
+class RateLimitMigrating(BaseException):
+    ...
+
+
+class RateLimit:
+    def __init__(self, time_offset: float = 0.3) -> None:
+        """Used to time gate a large batch of requests to only occur X every Y seconds. Used via ``async with``
+
+        NOT THREAD SAFE.
```

Comment:
How does this work with voice things? Iirc that's thread based in some aspects or at-least it used to be

Reply:
does voice use HTTP rate limits? I thought that was a websocket/gateway sort of thing.